### PR TITLE
Add: Local Refs

### PR DESCRIPTION
**Pull request by mattKorwel**
Tested this against the https://github.com/github/platform-corehub-reports/blob/main/reports.yml and it looks all good to me.

### CODE REVIEW

**Comment by mattKorwel** [comment]
Valdation output:

`
Building url for local ref 766 for repo https://github.com/github/ghes-infrastructure/issues
Building url for local ref 783 for repo https://github.com/github/ghes-infrastructure/issues
Building url for local ref 1061 for repo https://github.com/github/ghes-infrastructure/issues
Building url for local ref 1053 for repo https://github.com/github/ghes-infrastructure/issues
Building url for local ref 1197 for repo https://github.com/github/ghes-infrastructure/issues
Building url for local ref 1184 for repo https://github.com/github/ghes-infrastructure/issues
Building url for local ref 1183 for repo https://github.com/github/ghes-infrastructure/issues
Building url for local ref 955 for repo https://github.com/github/ghes-infrastructure/issues
Building url for local ref 825 for repo https://github.com/github/ghes-infrastructure/issues
Building url for local ref 976 for repo https://github.com/github/ghes-infrastructure/issues
Building url for local ref 1008 for repo https://github.com/github/ghes-infrastructure/issues
Building url for local ref 957 for repo https://github.com/github/ghes-infrastructure/issues
Building url for local ref 1126 for repo https://github.com/github/ghes-infrastructure/issues
Building url for local ref 1071 for repo https://github.com/github/ghes-infrastructure/issues
Building url for local ref 978 for repo https://github.com/github/ghes-infrastructure/issues
Building url for local ref 789 for repo https://github.com/github/ghes-infrastructure/issues
match: https://github.com/github/ghes-infrastructure/issues/812
/repos/github/ghes-infrastructure/issues/812 [200]rate : 4813/5000
/repos/github/ghes-infrastructure/issues/812/comments?per_page=100 [200]rate : 4812/5000
/repos/github/ghes-infrastructure/issues/812/events?per_page=100 [200]rate : 4811/5000
Preview mode only
Label already exists: > GHPI
Preview mode only
Label already exists: >> GHPI MVP

match: https://github.com/github/ghes-infrastructure/issues/788
/repos/github/ghes-infrastructure/issues/788 [200]rate : 4810/5000
/repos/github/ghes-infrastructure/issues/788/comments?per_page=100 [200]rate : 4809/5000
/repos/github/ghes-infrastructure/issues/788/events?per_page=100 [200]rate : 4808/5000
Preview mode only
Label already exists: > GHPI
Preview mode only
Label already exists: >> GHPI MVP

match: https://github.com/github/ghes-infrastructure/issues/832
/repos/github/ghes-infrastructure/issues/832 [200]rate : 4807/5000
/repos/github/ghes-infrastructure/issues/832/comments?per_page=100 [200]rate : 4806/5000
/repos/github/ghes-infrastructure/issues/832/events?per_page=100 [200]rate : 4805/5000
Preview mode only
Label already exists: > GHPI
Preview mode only
Label already exists: >> GHPI MVP

match: https://github.com/github/ghes-infrastructure/issues/813
/repos/github/ghes-infrastructure/issues/813 [200]rate : 4804/5000
/repos/github/ghes-infrastructure/issues/813/comments?per_page=100 [200]rate : 4803/5000
/repos/github/ghes-infrastructure/issues/813/events?per_page=100 [200]rate : 4802/5000
Preview mode only
Label already exists: > GHPI
Preview mode only
Label already exists: >> GHPI MVP

match: https://github.com/github/ghes-infrastructure/issues/959
/repos/github/ghes-infrastructure/issues/959 [200]rate : 4801/5000
/repos/github/ghes-infrastructure/issues/959/comments?per_page=100 [200]rate : 4800/5000
/repos/github/ghes-infrastructure/issues/959/events?per_page=100 [200]rate : 4799/5000
Preview mode only
Label already exists: > GHPI
Preview mode only
Label already exists: >> GHPI MVP

match: https://github.com/github/ghes-infrastructure/issues/846
/repos/github/ghes-infrastructure/issues/846 [200]rate : 4798/5000
/repos/github/ghes-infrastructure/issues/846/comments?per_page=100 [200]rate : 4797/5000
/repos/github/ghes-infrastructure/issues/846/events?per_page=100 [200]rate : 4796/5000
Preview mode only
Label already exists: > GHPI
Preview mode only
Label already exists: >> GHPI MVP

match: https://github.com/github/ghes-infrastructure/issues/847
/repos/github/ghes-infrastructure/issues/847 [200]rate : 4795/5000
/repos/github/ghes-infrastructure/issues/847/comments?per_page=100 [200]rate : 4794/5000
/repos/github/ghes-infrastructure/issues/847/events?per_page=100 [200]rate : 4793/5000
/repositories/208883147/issues/847/events?per_page=100&page=2 [200]rate : 4792/5000
Preview mode only
Label already exists: > GHPI
Preview mode only
Label already exists: >> GHPI MVP

match: https://github.com/github/ghes-infrastructure/issues/1061
/repos/github/ghes-infrastructure/issues/1061 [200]rate : 4791/5000
/repos/github/ghes-infrastructure/issues/1061/comments?per_page=100 [200]rate : 4790/5000
/repos/github/ghes-infrastructure/issues/1061/events?per_page=100 [200]rate : 4789/5000
Preview mode only
Label already exists: > GHPI
Preview mode only
Label already exists: >> GHPI MVP

match: https://github.com/github/ghes-infrastructure/issues/1041
/repos/github/ghes-infrastructure/issues/1041 [200]rate : 4788/5000
/repos/github/ghes-infrastructure/issues/1041/comments?per_page=100 [200]rate : 4787/5000
/repos/github/ghes-infrastructure/issues/1041/events?per_page=100 [200]rate : 4786/5000
Preview mode only
Label already exists: > GHPI
Preview mode only
Label already exists: >> GHPI MVP

match: https://github.com/github/ghes-infrastructure/issues/1056
/repos/github/ghes-infrastructure/issues/1056 [200]rate : 4785/5000
/repos/github/ghes-infrastructure/issues/1056/events?per_page=100 [200]rate : 4784/5000
Preview mode only
Label already exists: > GHPI
Preview mode only
Label already exists: >> GHPI MVP

match: https://github.com/github/ghes-infrastructure/issues/1145
/repos/github/ghes-infrastructure/issues/1145 [200]rate : 4783/5000
/repos/github/ghes-infrastructure/issues/1145/comments?per_page=100 [200]rate : 4782/5000
/repos/github/ghes-infrastructure/issues/1145/events?per_page=100 [200]rate : 4781/5000
Preview mode only
Label already exists: > GHPI
Preview mode only
Label already exists: >> GHPI MVP

match: https://github.com/github/enterprise2/pull/23219
/repos/github/enterprise2/issues/23219 [200]rate : 4780/5000
/repos/github/enterprise2/issues/23219/events?per_page=100 [200]rate : 4779/5000
Skipping.  Only processing with label feature

match: https://github.slack.com/archives/C014U6D3H6K/p1600900669050500
/repos/archives/C014U6D3H6K/issues/[404]
Ignoring invalid issue url: https://github.slack.com/archives/C014U6D3H6K/p1600900669050500
(Not Found)

match: https://github.com/github/ghes-infrastructure/issues/942
/repos/github/ghes-infrastructure/issues/942 [200]rate : 4777/5000
/repos/github/ghes-infrastructure/issues/942/comments?per_page=100 [200]rate : 4776/5000
/repos/github/ghes-infrastructure/issues/942/events?per_page=100 [200]rate : 4775/5000
Skipping.  Only processing with label feature

match: https://github.com/github/ghes-infrastructure/issues/941
/repos/github/ghes-infrastructure/issues/941 [200]rate : 4774/5000
/repos/github/ghes-infrastructure/issues/941/comments?per_page=100 [200]rate : 4773/5000
/repos/github/ghes-infrastructure/issues/941/events?per_page=100 [200]rate : 4772/5000
Skipping.  Only processing with label feature

match: https://github.com/github/ghes-infrastructure/issues/766
/repos/github/ghes-infrastructure/issues/766 [200]rate : 4771/5000
/repos/github/ghes-infrastructure/issues/766/comments?per_page=100 [200]rate : 4770/5000
/repos/github/ghes-infrastructure/issues/766/events?per_page=100 [200]rate : 4769/5000
Preview mode only
Label already exists: > GHPI
Preview mode only
Label already exists: >> GHPI MVP

match: https://github.com/github/ghes-infrastructure/issues/783
/repos/github/ghes-infrastructure/issues/783 [200]rate : 4768/5000
/repos/github/ghes-infrastructure/issues/783/comments?per_page=100 [200]rate : 4767/5000
/repos/github/ghes-infrastructure/issues/783/events?per_page=100 [200]rate : 4766/5000
Preview mode only
Label already exists: > GHPI
Preview mode only
Label already exists: >> GHPI MVP

match: https://github.com/github/ghes-infrastructure/issues/1061
/repos/github/ghes-infrastructure/issues/1061 [304, cache]
/repos/github/ghes-infrastructure/issues/1061/comments?per_page=100 [304, cache]
/repos/github/ghes-infrastructure/issues/1061/events?per_page=100 [304, cache]
Preview mode only
Label already exists: > GHPI
Preview mode only
Label already exists: >> GHPI MVP

match: https://github.com/github/ghes-infrastructure/issues/1053
/repos/github/ghes-infrastructure/issues/1053 [200]rate : 4765/5000
/repos/github/ghes-infrastructure/issues/1053/events?per_page=100 [200]rate : 4764/5000
Skipping.  Only processing with label feature

match: https://github.com/github/ghes-infrastructure/issues/1197
/repos/github/ghes-infrastructure/issues/1197 [200]rate : 4763/5000
/repos/github/ghes-infrastructure/issues/1197/comments?per_page=100 [200]rate : 4762/5000
/repos/github/ghes-infrastructure/issues/1197/events?per_page=100 [200]rate : 4761/5000
Skipping.  Only processing with label feature

match: https://github.com/github/ghes-infrastructure/issues/1184
/repos/github/ghes-infrastructure/issues/1184 [200]rate : 4760/5000
/repos/github/ghes-infrastructure/issues/1184/comments?per_page=100 [200]rate : 4759/5000
/repos/github/ghes-infrastructure/issues/1184/events?per_page=100 [200]rate : 4758/5000
Skipping.  Only processing with label feature

match: https://github.com/github/ghes-infrastructure/issues/1183
/repos/github/ghes-infrastructure/issues/1183 [200]rate : 4757/5000
/repos/github/ghes-infrastructure/issues/1183/comments?per_page=100 [200]rate : 4756/5000
/repos/github/ghes-infrastructure/issues/1183/events?per_page=100 [200]rate : 4755/5000
Skipping.  Only processing with label feature

match: https://github.com/github/ghes-infrastructure/issues/955
/repos/github/ghes-infrastructure/issues/955 [200]rate : 4754/5000
/repos/github/ghes-infrastructure/issues/955/comments?per_page=100 [200]rate : 4753/5000
/repos/github/ghes-infrastructure/issues/955/events?per_page=100 [200]rate : 4752/5000
Skipping.  Only processing with label feature

match: https://github.com/github/ghes-infrastructure/issues/825
/repos/github/ghes-infrastructure/issues/825 [200]rate : 4751/5000
/repos/github/ghes-infrastructure/issues/825/comments?per_page=100 [200]rate : 4750/5000
/repos/github/ghes-infrastructure/issues/825/events?per_page=100 [200]rate : 4749/5000
Skipping.  Only processing with label feature

match: https://github.com/github/ghes-infrastructure/issues/976
/repos/github/ghes-infrastructure/issues/976 [200]rate : 4748/5000
/repos/github/ghes-infrastructure/issues/976/events?per_page=100 [200]rate : 4747/5000
Skipping.  Only processing with label feature

match: https://github.com/github/ghes-infrastructure/issues/1008
/repos/github/ghes-infrastructure/issues/1008 [200]rate : 4746/5000
/repos/github/ghes-infrastructure/issues/1008/events?per_page=100 [200]rate : 4745/5000
Skipping.  Only processing with label feature

match: https://github.com/github/ghes-infrastructure/issues/957
/repos/github/ghes-infrastructure/issues/957 [200]rate : 4744/5000
/repos/github/ghes-infrastructure/issues/957/comments?per_page=100 [200]rate : 4743/5000
/repos/github/ghes-infrastructure/issues/957/events?per_page=100 [200]rate : 4742/5000
Skipping.  Only processing with label feature

match: https://github.com/github/ghes-infrastructure/issues/1126
/repos/github/ghes-infrastructure/issues/1126 [200]rate : 4741/5000
/repos/github/ghes-infrastructure/issues/1126/events?per_page=100 [200]rate : 4740/5000
Skipping.  Only processing with label feature

match: https://github.com/github/ghes-infrastructure/issues/1071
/repos/github/ghes-infrastructure/issues/1071 [200]rate : 4739/5000
/repos/github/ghes-infrastructure/issues/1071/comments?per_page=100 [200]rate : 4738/5000
/repos/github/ghes-infrastructure/issues/1071/events?per_page=100 [200]rate : 4737/5000
Skipping.  Only processing with label feature

match: https://github.com/github/ghes-infrastructure/issues/978
/repos/github/ghes-infrastructure/issues/978 [200]rate : 4736/5000
/repos/github/ghes-infrastructure/issues/978/events?per_page=100 [200]rate : 4735/5000
Skipping.  Only processing with label feature

match: https://github.com/github/ghes-infrastructure/issues/789
/repos/github/ghes-infrastructure/issues/789 [200]rate : 4734/5000
/repos/github/ghes-infrastructure/issues/789/events?per_page=100 [200]rate : 4733/5000
Skipping.  Only processing with label feature`

File: processors/project-reference-labeler.ts
Comment (bryanmacfarlane) [inline processors/project-reference-labeler.ts:116]: probably want let here instead of var

File: processors/project-reference-labeler.ts
Comment (bryanmacfarlane) [inline processors/project-reference-labeler.ts:123]: can localRefs be null?  Have we tested with issues without local refs?

File: processors/project-reference-labeler.ts
Comment (bryanmacfarlane) [inline processors/project-reference-labeler.ts:121]: path.dirname will do this:

```javascript
> path.dirname("https://site.com/foo/bar/1234")
'https://site.com/foo/bar'
```

File: processors/project-reference-labeler.ts
Comment (bryanmacfarlane) [inline processors/project-reference-labeler.ts:124]: path.join will do this:

```javascript
> path.join("https://site.com/foo/bar", "1234")
'https:/site.com/foo/bar/1234'
```

File: Dockerfile
Comment (bryanmacfarlane) [inline Dockerfile:7]: I like the idea of using docker here for a consistent local dev run.  However, If we're going to do this, I would prefer for there to be just one way and integrate it into ./interactive.sh (which is in the contribute.md).

Perhaps we can pull this out into a separate PR and not hold up the needed local ref changes below?

Comment (mattKorwel) [reply Dockerfile:7]: sure. I have no problem pulling out the docker stuff. I just needed it as I don't like polluting my machine with node :)

in my case I was doing ./docker.build.sh then ./docker.dev.sh and then running all the commands, including interactive in the container.  your call though. if you want it out, lets remove it.

File: processors/project-reference-labeler.ts
Comment (mattKorwel) [reply processors/project-reference-labeler.ts:124]: ack.

File: processors/project-reference-labeler.ts
Comment (mattKorwel) [reply processors/project-reference-labeler.ts:123]: tested with out refs and it totally works fine....

File: processors/project-reference-labeler.ts
Comment (mattKorwel) [reply processors/project-reference-labeler.ts:116]: ack

File: Dockerfile
Comment (bryanmacfarlane) [reply Dockerfile:7]: yeah, I love the idea of this but let's take in separate PR and merge with interactive.sh

**Comment by mattKorwel** [comment]
@bryanmacfarlane we should be gtg when you are.

**Review by bryanmacfarlane — APPROVED** [review]

**Comment by bryanmacfarlane** [comment]
awesome.  thanks for this contribution

**Pull request merged**
